iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -308,16 +418,47 @@ - (void)messageViewControllerWasDismissed {
             [self displayMessage:self.messageDisplayQueue.firstObject];
             return;
         } else {
-            // Hide the window and call makeKeyWindow to ensure the IAM will not be shown
-            self.window.hidden = true;
-            [UIApplication.sharedApplication.delegate.window makeKeyWindow];
-            
+            [self hideWindow];
             // Evaulate any IAMs (could be new IAM or added trigger conditions)
             [self evaluateMessages];
         }
     }
 }
 
+/*
+ Hide the window and call makeKeyWindow to ensure the IAM will not be shown
+ */
+- (void)hideWindow {
+    self.window.hidden = true;
+    [UIApplication.sharedApplication.delegate.window makeKeyWindow];
+}
+
+- (void)persistInAppMessageForRedisplay:(OSInAppMessage *)message {
+    // If the IAM doesn't have the re display prop there is no need to save it
+    if (![message.displayStats isRedisplayEnabled])
+      return;
+
+    let displayTimeSeconds = self.dateGenerator();
+    message.displayStats.lastDisplayTime = displayTimeSeconds;
+    [message.displayStats incrementDisplayQuantity];
+    message.isTriggerChanged = false;
+    message.isDisplayedInSession = true;
+
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"redisplayedInAppMessages: %@", [_redisplayedInAppMessages description]]];
+    
+    // Update the data to enable future re displays
+    // Avoid calling the userdefault data again
+    [_redisplayedInAppMessages setObject:message forKey:message.messageId];
+    
+    [OneSignalUserDefaults.initStandard saveDictionaryForKey:OS_IAM_REDISPLAY_DICTIONARY withValue:self.redisplayedInAppMessages];
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"persistInAppMessageForRedisplay: %@ \nredisplayedInAppMessages: %@", [message description], [_redisplayedInAppMessages description]]];
+}
+
+- (BOOL)isClickAvailable:(OSInAppMessage *)message withClickId:(NSString *)clickId {
+    // If IAM has redisplay the clickId may be available
+    return !([message.displayStats isRedisplayEnabled] && [message isClickAvailable:clickId]) && [self.clickedClickIds containsObject:clickId];
```

Comment:
Isn't `[message isClickAvailable:clickId]` same as `![self.clickedClickIds containsObject:clickId]`?

Are `self.clickedClickIds` and `_clickedClickIds` the same set?